coresimd/src/x86/i686/sse42.rs

```diff
@@ -10,22 +12,21 @@ use stdsimd_test::assert_instr;
 #[inline(always)]
 #[target_feature = "+sse4.2"]
 #[cfg_attr(test, assert_instr(pcmpgtq))]
-pub unsafe fn _mm_cmpgt_epi64(a: i64x2, b: i64x2) -> i64x2 {
-    a.gt(b)
+pub unsafe fn _mm_cmpgt_epi64(a: __m128i, b: __m128i) -> __m128i {
+    mem::transmute(simd_gt::<_, i64x2>(a.as_i64x2(), b.as_i64x2()))
 }
 
 #[cfg(test)]
 mod tests {
-    use v128::*;
-    use x86::i686::sse42;
+    use x86::*;
 
     use stdsimd_test::simd_test;
 
     #[simd_test = "sse4.2"]
-    unsafe fn _mm_cmpgt_epi64() {
-        let a = i64x2::splat(0x00).replace(1, 0x2a);
-        let b = i64x2::splat(0x00);
-        let i = sse42::_mm_cmpgt_epi64(a, b);
-        assert_eq!(i, i64x2::new(0x00, 0xffffffffffffffffu64 as i64));
+    unsafe fn test_mm_cmpgt_epi64() {
+        let a = _mm_setr_epi64x(0, 0x2a);
```

Comment:
which intrinsic is this?

Reply:
Heh a good question! All types so far have had an `_mm_set_XX` constructor which sets all the lanes, and *most* have a `setr` which is the opposite order of specifying the lanes. The existing methods like `i64x2::new` are the same order as the `setr` intrinsics, so I've been mechanically changing `foo::new` to `_mm_setr_foo`. Unfortunately though `i64x2`'s equivalent *only* has `_mm_set_epi64x` and doesn't have the `setr` equivalent, so I just [vendored it](https://github.com/rust-lang-nursery/stdsimd/blob/master/coresimd/src/x86/test.rs#L34-L39) to allow easy translation.

It should maybe have a name that indicates it's not an intrinsic though...

Reply:
(note that the intrinsic here is only defined for tests, not exported from the library)